api/envoy/api/v2/auth/cert.proto

```diff
@@ -219,6 +219,16 @@ message CertificateValidationContext {
 
   // If specified, Envoy will not reject expired certificates.
   bool allow_expired_certificate = 8;
+
```

Comment:
Can you update the description at https://github.com/envoyproxy/envoy/blob/e7d71aa3c3afe20d2a34572edd12c108f05e497d/api/envoy/api/v2/auth/cert.proto#L118 to describe how all these options interact? I share @PiotrSikora that we might have some redundancy here, in particular I think we could collapse the request/require client certificate fields with the right combination of options.